From dc2ec72a127c75ef0614056b3f319607b4b4414b Mon Sep 17 00:00:00 2001
From: Elliott Mitchell <ehem+android_device_lge_v20-common@m5p.com>
Date: Mon, 26 Feb 2018 23:03:38 -0800
Subject: [PATCH 2/2] Build fixes for V20 second screen code

With these changes this code now builds in LineageOS.  These look to be
the correct fixes, hopefully this gives us something to really play with.
---
 .../android/systemui/secondscreen/SecondScreenActivity.java  | 12 ++++++------
 .../com/android/systemui/secondscreen/SecondScreenText.java  |  7 +++++--
 .../android/systemui/secondscreen/SecondScreenToggles.java   |  7 +++++--
 3 files changed, 16 insertions(+), 10 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenActivity.java b/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenActivity.java
index 3fab39e..84e7ccb 100644
--- a/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenActivity.java
+++ b/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenActivity.java
@@ -1,18 +1,18 @@
 package com.android.systemui.secondscreen;
 
 import android.app.Activity;
 import android.support.v4.app.FragmentActivity;
 import android.support.v4.app.FragmentManager;
 import android.support.v4.app.Fragment;
 import android.support.v4.view.ViewPager;
-import android.support.v13.app.FragmentPagerAdapter;
+import android.support.v4.app.FragmentPagerAdapter;
 import android.content.Context;
 import android.content.Intent;
 import android.graphics.Color;
 import android.graphics.PixelFormat;
 import android.os.Bundle;
 import android.view.Gravity;
 import android.view.LayoutInflater;
 import android.view.MotionEvent;
 import android.view.View;
 import android.view.ViewGroup;
@@ -38,21 +38,21 @@ public class SecondScreenActivity extends FragmentActivity implements OnTouchLis
 
         inflater = (LayoutInflater)this.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
 
         wm = (WindowManager) getSystemService(Context.WINDOW_SERVICE);
 
         rootView = (LinearLayout) inflater.inflate(R.layout.second_screen_root, null);
         rootView.setBackgroundColor(Color.BLACK);
         rootView.setOnTouchListener(this);
 
         viewPager = (ViewPager) rootView.findViewById(R.id.second_screen_viewpager);
-        viewsPagerAdapter = new ViewsPagerAdapter(getFragmentManager());
+        viewsPagerAdapter = new ViewsPagerAdapter(getSupportFragmentManager());
         viewPager.setAdapter(viewsPagerAdapter);
 
         WindowManager.LayoutParams params = new WindowManager.LayoutParams(WindowManager.LayoutParams.WRAP_CONTENT, WindowManager.LayoutParams.WRAP_CONTENT, WindowManager.LayoutParams.TYPE_SYSTEM_ERROR, WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE | WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL | WindowManager.LayoutParams.FLAG_LAYOUT_IN_SCREEN, PixelFormat.TRANSLUCENT);
         params.gravity = Gravity.LEFT | Gravity.TOP;
         params.x = 400;
         params.y = 0;
         params.width = 1040;
         params.height = 160;
         wm.addView(rootView, params);
 
@@ -66,35 +66,35 @@ public class SecondScreenActivity extends FragmentActivity implements OnTouchLis
             rootView = null;
         }
     }
 
     @Override
     public boolean onTouch(View v, MotionEvent event) {
         System.out.println("SecondScreenTouch");
         return false;
     }
 
-    class ViewsPagerAdapter extends FragmentPagerAdapter {
+    public static class ViewsPagerAdapter extends FragmentPagerAdapter {
         private ArrayList<Fragment> frags = new ArrayList<>();
-        
+
         public ViewsPagerAdapter(FragmentManager fm) {
             super(fm);
             frags.add((Fragment) new SecondScreenText());
             frags.add((Fragment) new SecondScreenToggles());
         }
 
         @Override
         public Fragment getItem(int position) {
             return (Fragment) frags.get(position);
         }
 
         @Override
         public CharSequence getPageTitle(int position) {
-            return "";*
+            return "";
         }
 
         @Override
         public int getCount() {
             return frags.size();
         }
-    }   
+    }
 }
diff --git a/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenText.java b/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenText.java
index 39be912..0be8714 100644
--- a/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenText.java
+++ b/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenText.java
@@ -1,21 +1,24 @@
 package com.android.systemui.secondscreen;
 
 import android.graphics.Color;
 import android.os.Bundle;
 import android.support.v4.app.Fragment;
+import android.view.View;
 import android.view.LayoutInflater;
 import android.view.ViewGroup;
 import android.widget.TextView;
 
+import com.android.systemui.R;
+
 public class SecondScreenText extends Fragment {
 
     @Override
-    private View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         ViewGroup rootView = (ViewGroup) inflater.inflate(R.layout.second_screen_text, container, false);
         TextView text = (TextView) rootView.findViewById(R.id.second_screen_text);
         text.setText("V20");
         text.setTextColor(Color.WHITE);
-        return rootview;
+        return rootView;
     }
 
 }
diff --git a/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenToggles.java b/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenToggles.java
index d68c28b..bbb7a27 100644
--- a/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenToggles.java
+++ b/packages/SystemUI/src/com/android/systemui/secondscreen/SecondScreenToggles.java
@@ -1,17 +1,20 @@
 package com.android.systemui.secondscreen;
 
 import android.graphics.Color;
 import android.os.Bundle;
 import android.support.v4.app.Fragment;
+import android.view.View;
 import android.view.LayoutInflater;
 import android.view.ViewGroup;
 
+import com.android.systemui.R;
+
 public class SecondScreenToggles extends Fragment {
 
     @Override
-    private View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         ViewGroup rootView = (ViewGroup) inflater.inflate(R.layout.second_screen_toggles, container, false);
-        return rootview;
+        return rootView;
     }
 
 }
-- 
2.1.4

